internal/fsutil/create.go

```diff
@@ -162,6 +183,30 @@ func createSymlink(o *CreateOptions) error {
 	return os.Symlink(o.Link, o.Path)
 }
 
+var ErrLinkTargetNotExist = fmt.Errorf("link target does not exist")
+
+func createHardLink(o *CreateOptions) error {
+	debugf("Creating hard link: %s => %s", o.Path, o.Link)
+	linkInfo, err := os.Lstat(o.Link)
```

Comment:
The `ErrLinkTargetNotExist` is created to let the `extract` handle the case when a hard link is extracted alone without explicitly comparing the error string.

Reply:
We still need to `Lstat` both files if we are to use `os.SameFile`, as it takes two `os.FileInfo` as it parameters.

Reply:
The difference is that we do not need to `Lstat` every time, only when the target already exist, that is two less syscalls per file creation by default. 

`ErrLinkTargetNotExist` is not needed because we can return the actual error and use `os.IsExist`.